paddle/math/Matrix.cpp

```diff
@@ -2005,17 +2021,30 @@ void CpuMatrix::maxPoolForward(Matrix& inputMat,
           int wstart = pw * strideW - paddingW;
           int wend = std::min(wstart + sizeX, imgSizeW);
           wstart = std::max(wstart, 0);
-          for (int h = hstart; h < hend; ++h) {
-            for (int w = wstart; w < wend; ++w) {
-              outData[ph * outputW + pw] = std::max(
-                  outData[ph * outputW + pw], inputData[h * imgSizeW + w]);
+          if (maskMatP == NULL) {
```

Comment:
Is the `if (maskMatP == NULL)` can be inside the for loop? And use `maskData == NULL` instead of `maskMatP == NULL`.

Reply:
Done